SofaKernel/modules/SofaEigen2Solver/src/SofaEigen2Solver/SVDLinearSolver.cpp

```diff
@@ -57,22 +58,25 @@ void SVDLinearSolver<TMatrix,TVector>::solve(Matrix& M, Vector& x, Vector& b)
     const bool verbose  = f_verbose.getValue();
 
     /// Convert the matrix and the right-hand vector to Eigen objects
+    sofa::helper::AdvancedTimer::stepBegin("convertToEigen");
     Eigen::MatrixXd m(M.rowSize(),M.colSize());
     Eigen::VectorXd rhs(M.rowSize());
     for(unsigned i=0; i<(unsigned)M.rowSize(); i++ )
     {
         for( unsigned j=0; j<(unsigned)M.colSize(); j++ )
-            m(i,j) = M[i][j];
+            m(i,j) = M.element(i, j);
         rhs(i) = b[i];
     }
+    sofa::helper::AdvancedTimer::stepEnd("convertToEigen");
 
     msg_info_when(verbose) << "solve, Here is the matrix m:  "
                            << m ;
 
     /// Compute the SVD decomposition and the condition number
+    sofa::helper::AdvancedTimer::stepBegin("SVDDecomposition");
     Eigen::JacobiSVD<Eigen::MatrixXd> svd(m, Eigen::ComputeThinU | Eigen::ComputeThinV);
     f_conditionNumber.setValue( (Real)(svd.singularValues()(0) / svd.singularValues()(M.rowSize()-1)) );
-
+    sofa::helper::AdvancedTimer::stepEnd("SVDDecomposition");
```

Comment:
In general is is better to use the RAII version of the timer.
See: ScopedAdvancedTimer;

Reply:
In general yes, but it sometimes requires to change the organisation of the code if an object must live in two different scopes